public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -287,7 +320,9 @@ export const GeneratePopoverBody: React.FC<{
                 aria-label="back-to-summary"
                 size="m"
                 onClick={() => {
-                  setShowInsight(false);
+                  if (insight !== '') {
```

Comment:
Q: why adding this check? Shouldn't it always return back to summary when user click the return button?

Reply:
I add this check to prevent user click the button multiple times to switch between summary/insights and result in multiple request, is this appropriate?

Reply:
This is to ensure the user doesn't lick on the arrow button multiple times since the button is a part of the popover header that shows in the loading state, would this be appropriate? Or I can hide the button in loading state through.